Emby.Server.Implementations/ApplicationHost.cs

```diff
@@ -1150,6 +1150,18 @@ public string GetSmartApiUrl(IPAddress remoteAddr, int? port = null)
         /// <inheritdoc/>
         public string GetSmartApiUrl(HttpRequest request, int? port = null)
         {
+            // Return the host in the HTTP request as the API url
+            if (ConfigurationManager.GetNetworkConfiguration().EnablePublishedServerUriByRequest)
+            {
+                int? requestPort = request.Host.Port;
+                if ((requestPort == 80 && request.Scheme == "http") || (requestPort == 443 && request.Scheme == "https"))
```

Comment:
If `Scheme` is always lowercase:
```suggestion
                if ((requestPort == 80 && string.Equals(request.Scheme, "http", StringComparison.Ordinal)) || (requestPort == 443 && string.Equals(request.Scheme, "https", StringComparison.Ordinal)))
```

Otherwise:
```suggestion
                if ((requestPort == 80 && string.Equals(request.Scheme, "http", StringComparison.OrdinalIgnoreCase)) || (requestPort == 443 && string.Equals(request.Scheme, "https", StringComparison.OrdinalIgnoreCase)))
```

Reply:
Is string.Equals preferred over == ?

Reply:
Yes

Reply:
Alright, fixed